components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java

```diff
@@ -41,19 +43,32 @@
 import org.wso2.carbon.user.core.listener.UserOperationEventListener;
 import org.wso2.carbon.user.core.service.RealmService;
 
+import static org.wso2.carbon.identity.oauth.Error.FORBIDDEN;
+
 /**
  * OAuth OSGi service component.
  */
 @Component(
         name = "identity.oauth.component",
         immediate = true
 )
+@Deprecated
 public class OAuthServiceComponent {
 
     private static final Log log = LogFactory.getLog(OAuthServiceComponent.class);
     private ServiceRegistration serviceRegistration = null;
+    public static final String ENABLE_OAUTH1_CONFIG = "LegacyFeatures.EnableOAuth1";
+
+    protected void activate(ComponentContext context) throws IdentityOAuthClientException {
+
+        boolean isOAuth1Enabled = Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_OAUTH1_CONFIG));
+        if (!isOAuth1Enabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("OAuth 1.0 Endpoint is disabled.");
+            }
+            throw new IdentityOAuthClientException(FORBIDDEN.getErrorCode(), "OAuth 1.0 is not supported");
```

Comment:
this is not needed. We don't need to disable the OSGi service component.

Reply:
Corrected https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/e5dca5f8761be050156782479f65a2d611f76780

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/e5dca5f8761be050156782479f65a2d611f76780